Scripts/EasternKingdoms: Ragged John

mysterious-static opened the pull request:
**Changes proposed:**

- Migrates Ragged John to SAI
- Removes zone_burning_steppes.cpp

**Target branch(es):** 3.3.5

**Tests performed:**
Builds. Verified SQL script was syntactically correct.

**Known issues and TODO list:** (add/remove lines as needed)

- [x] The included SAI is a direct port from C++. I still need to verify that the event that I noted in SQL comments will actually fire or whether that also needs to be scripted. Will be doing this tonight.

Commit:
migrate Ragged John to SAI

Commit:
Update 9999_99_99_99_world.sql

Commit:
Update 9999_99_99_99_world.sql

Commit:
add notes on missing events for this NPC

Will test and remove notes before PR.

Aokromes commented on src/server/scripts/EasternKingdoms/eastern_kingdoms_script_loader.cpp:
```diff
@@ -175,7 +175,7 @@ void AddSC_instance_zulgurub();
 void AddSC_arathi_highlands();
 void AddSC_blasted_lands();
-void AddSC_burning_steppes();
+void AddSC_duskwood();
```
duplicate row.

velinath replied:
How in the world did I manage that?

Commit:
Remove introduced typo

offl commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,20 @@
+
+-- per wowhead - player dialog w/NPC "Milk me, John." "With pleasure... just let me savor this moment for a minute longer." "Do it... do it now."
+-- verifiable w/ broadcast_text id 5833, 5834, 5835
+-- no video available that I can find, unfortunately, so I have no idea how this relates at all to the actual event action above, or whether that may be handled by the spell (I'll check on this before PR is opened.)
```
```sql
SELECT * FROM `broadcast_text` WHERE `ID` BETWEEN 5833 AND 5835;
```
```
    ID  Language  MaleText                                                             FemaleText           
------  --------  -------------------------------------------------------------------  ---------------------
  5833         0  Milk me, John.                                                       Milk me, John.       
  5834         0  With pleasure... just let me savor this moment for a minute longer.                       
  5835         0  Do it... Do it now.                                                  Do it... Do it now.  
```

Quite possible he should cast the spell only after player select the gossip.

velinath replied:
Could be gossip, in which case this whole thing becomes a lot easier. I figured based on that Wowhead comment it was scripted /say text - just based on the word "dialogue" - but I was hoping someone would have a video somewhere - without other sources, I usually default to doing a direct conversion from the C++ script.

I'm fine with moving it to gossip if devs are also comfortable with that - maybe we can get a check in from them on that :)

offl replied:
One text already present in npc_text, so..
```sql
SELECT * FROM `npc_text` WHERE `ID`=3496;
```
```text
    ID  text0_0                                                              
------  ---------------------------------------------------------------------
  3496  With pleasure... just let me savor this moment for a minute longer.  
```

velinath replied:
good catch, that's enough for me to be going on with! I'll make necessary changes sometime today or tomorrow.

Commit:
First pass on rewrite, thanks @offl for research

I'll revisit this later today but I wanted to get initial stuff down while it was fresh in my head.

Commit:
first update, more to come

sai still doesn't trigger properly. probably a super easy fix, I just don't have time, wanted to at least fix the conditions now

Commit:
Fixes wrong smart script id

The cast still isn't going off, maybe it needs castflags? Further testing required zzz

velinath commented:
actually this is fine now, i was running into a phantom problem because i had leftover conditions for smart scripts to fire

this is ready to go, tested ingame

Aokromes committed:
Rename 9999_99_99_99_world.sql to 2017_11_08_02_world.sql
